File: common/src/main/java/dev/obscuria/fragmentum/content/registry/DeferredBlock.java
package dev.obscuria.fragmentum.content.registry;

import net.minecraft.core.Holder;
import net.minecraft.world.level.block.Block;
import net.minecraft.world.level.block.state.BlockState;

import java.util.function.Supplier;

@SuppressWarnings("unused")
public class DeferredBlock<T extends Block> extends Deferred<Block, T> {

    public DeferredBlock(Supplier<Holder<Block>> supplier) {
        super(supplier);
    }

    public BlockState instantiate() {
        return get().defaultBlockState();
    }
}

File: common/src/main/kotlin/dev/obscuria/fragmentum/registry/DeferredBlock.kt
@file:Suppress("unused", "RedundantNullableReturnType")

package dev.obscuria.fragmentum.registry

import net.minecraft.core.Holder
import net.minecraft.world.level.block.Block
import net.minecraft.world.level.block.state.BlockState

class DeferredBlock<T : Block>(
    supplier: () -> Holder<Block>
) : Deferred<Block, T>(supplier) {

    fun instantiate(): BlockState = get().defaultBlockState()
}